src/backend/distributed/executor/multi_executor.c

```diff
@@ -246,6 +256,12 @@ CitusExecutorRun(QueryDesc *queryDesc,
 		executorBoundParams = savedBoundParams;
 		ExecutorLevel--;
 
+		/*
+		 * In case of an exception, reset the pinned shard-key, for more
+		 * details see the function header.
+		 */
+		ResetAllowedShardKeyValue();
```

Comment:
This should probably have an ExecutorLevel == 0 check, otherwise we'll reset it when the pl/pgsql catches an exception (which uses savepoints).

Reply:
I previously had it, somehow, I removed it in the new PR